src/fellows.ts

```diff
@@ -14,44 +22,73 @@ export type FellowObject = {
 export const fetchAllFellows = async (
   logger: ActionLogger,
 ): Promise<FellowObject[]> => {
-  let api: ApiPromise;
-  logger.debug("Connecting to collective parachain");
-  // we connect to the collective rpc node
-  const wsProvider = new WsProvider(
-    "wss://polkadot-collectives-rpc.polkadot.io",
-  );
-  api = await ApiPromise.create({ provider: wsProvider });
+  logger.info("Initializing smoldot");
+  const smoldot = start();
+  const SmProvider = getSmProvider(smoldot);
+  let polkadotClient: ReturnType<typeof createClient> | null = null;
+
   try {
-    // We fetch all the members
-    const membersObj = await api.query.fellowshipCollective.members.entries();
+    const relayChain = await smoldot.addChain({
+      chainSpec: polkadot,
+    });
+    logger.debug("Connecting to collective parachain");
+    await smoldot.addChain({
+      chainSpec: polkadot_collectives,
+      potentialRelayChains: [relayChain],
+    });
+
+    logger.info("Initializing PAPI");
+    polkadotClient = createClient(
+      getChain({
+        provider: SmProvider({
+          potentialRelayChains: [relayChain],
+          chainSpec: polkadot_collectives,
+        }),
+        keyring: [],
+      }),
+    );
+
+    // We fetch all the members from the collective
+    const collectivesApi = polkadotClient.getTypedApi(collectiveDescriptor);
+    const memberEntries =
+      await collectivesApi.query.FellowshipCollective.Members.getEntries();
 
     // We iterate over the fellow data and convert them into usable values
     const fellows: FellowData[] = [];
-    for (const [key, rank] of membersObj) {
-      // @ts-ignore
-      const [address]: [string] = key.toHuman();
-      fellows.push({ address, ...(rank.toHuman() as object) } as FellowData);
+    for (const member of memberEntries) {
+      const [address] = member.keyArgs;
+      fellows.push({ address, rank: member.value } as FellowData);
```

Comment:
small nit:

I'm guessing that the reason why you are doing `as FellowData` is because in the type definition of `FellowData`, the `address` is a `string`, rather than a `SS58String`. If that's the case, then I would consider making one of the following changes:

- Option one (what I would do), define `FellowData` as:

```ts
interface FellowData {
  address: SS58String, rank: number
}
```

- Option 2, if for some obscure reason you don't want to change the type of `FellowData`, then simply do:

```ts
const address: string = member.keyArgs[0];
```

So that in the next line you don't have to cast it to `FellowData`.

Generally speaking, only use `as Whatever` if your really are out of options.

Reply:
Thanks for the review! That was a residual data from the time I was using pokladot-js.

I just tried without the `as` and it works out of the box!